src/manufacturer/manufacturerDialog.component.tsx

```diff
@@ -10,79 +14,51 @@ import {
   TextField,
   Typography,
 } from '@mui/material';
-
-import React from 'react';
-
 import {
   AddManufacturer,
-  Manufacturer,
   EditManufacturer,
   ErrorParsing,
-  ManufacturerDetails,
+  Manufacturer,
 } from '../app.types';
-import { useAddManufacturer, useEditManufacturer } from '../api/manufacturer';
 import { AxiosError } from 'axios';
+import { useAddManufacturer, useEditManufacturer } from '../api/manufacturer';
 import handleIMS_APIError from '../handleIMS_APIError';
-import { trimStringValues } from '../utils';
+
+const manufacturerSchema = z.object({
+  name: z.string().trim().min(1, { message: 'Please enter a name.' }),
+  url: z
+    .string()
+    .url({ message: 'Please enter a valid URL' })
```

Comment:
When editing and removing the URL, it gives
![image](https://github.com/ral-facilities/inventory-management-system/assets/90245114/950bf9b1-6ea9-4acb-978d-c812f3523324)

I would personally do something like
const transformEmptyStringToNull = (value: string) =>
  value === '' ? null : value;

const ZodStringOrNull = z.string().trim().transform(transformEmptyStringToNull);
const ZodUrlOrNull = ZodStringOrNull.transform((value) =>
  z
    .string()
    .url({ message: 'Please enter a valid URL' })
    .nullable()
    .parse(value)
);

So they can all be reused. This for the url also means it does the trim/null comparison first so a url of "    " would also be properly accounted for.

Reply:
The issue with this is error message. The error message are tailored to each text field